api-controllers/ServiceFabrikApiController.js

```diff
@@ -434,21 +434,24 @@ class ServiceFabrikApiController extends FabrikBaseController {
           backup_guid: backupGuid,
           tenant_id: tenantId
         };
-        return timeStamp ?
-          this.backupStore
-          .listBackupsOlderThan(backupFileOptions, new Date(timeStamp))
-          .then(oldBackupArray =>
-            _.sortBy(oldBackupArray, ['started_at']))
-          .then(sortedArray =>
-            _.findLast(sortedArray, backup => backup.state ===  CONST.OPERATION.SUCCEEDED))
-          .tap(successfulBackup => {
-            if (_.isEmpty(successfulBackup)) {
-              throw new NotFound(`No backup successful found for service instance '${instanceId}' before time_stamp ${new Date(timeStamp)} `);
-            }
-          }) :
-          this.backupStore.getBackupFile(backupFileOptions);
+        if (timeStamp) {
+          return this.backupStore
+            .listBackupsOlderThan(backupFileOptions, new Date(timeStamp))
+            .then(oldBackups =>
+              _.sortBy(oldBackups, ['started_at']))
```

Comment:
unnecessary sort, `listBackupsOlderThan` already has sort in predicate on `started_at`. please check.

Reply:
Wanted to be independent at this part, so did once more. As this part  is critical for quota check. Also did in filtering old backups.

Reply:
why dont u just add a unit test to `listBackupsOlderThan` which always returns sorted in ascending that way if someone changes that the Unit test fails. These are within the same component so really dont need to be this independent & end up doing repetitive processing across files in the same code flow.

Reply:
ok..will do.